ethers-providers/src/provider.rs

```diff
@@ -786,6 +786,151 @@ impl<P: JsonRpcClient> Middleware for Provider<P> {
         self.query_resolver(ParamType::String, &ens_name, ens::NAME_SELECTOR).await
     }
 
+    /// Returns the avatar HTTP link of the avatar that the `ens_name` resolves to (or None
+    /// if not configured)
+    ///
+    /// # Panics
+    ///
+    /// If the bytes returned from the ENS registrar/resolver cannot be interpreted as
+    /// a string. This should theoretically never happen.
+    async fn resolve_avatar(&self, ens_name: &str) -> Result<Url, ProviderError> {
+        let field = self.resolve_field(ens_name, "avatar").await?;
+        let url = Url::from_str(&field).map_err(|e| ProviderError::CustomError(e.to_string()))?;
+        let owner = self.resolve_name(ens_name).await?;
+        match url.scheme() {
+            "https" | "data" => Ok(url),
+            "ipfs" => erc::http_link_ipfs(url).map_err(ProviderError::CustomError),
+            "eip155" => {
+                let split: Vec<&str> = url.path().trim_start_matches("1/").split(':').collect();
+                let (inner_scheme, inner_path) = if split.len() == 2 {
+                    (split[0], split[1])
```

Comment:
this could be `str::split_once(':').ok_or_else(..)`, right?

Reply:
`split_once` won't fail if there are more of the split character, I feel like it's better to fail early? But because of the checks/deserialisation that happen right after, we wouldn't make invalid queries. So I'm on the fence